src/components/LogbookContext/index.tsx

```diff
@@ -639,6 +643,27 @@ export const LogbookProvider = ({
     }
   }
 
+  const getOwnersBalance = async (addresses: string[]) => {
+    const contract = new ethers.Contract(
+      env.contractAddress,
+      env.contractABI,
+      // ethers.getDefaultProvider(env.supportedChainId, { infura: env.infuraId })
+      new ethers.providers.InfuraProvider(env.supportedChainId, env.infuraId)
+    )
```

Comment:
I'm seeing more and more dup code in this `LogbookContext.Provider`, is time to change to a class based component?